python-package/lightgbm/sklearn.py

```diff
@@ -777,8 +772,10 @@ def _get_meta_data(collection, name, i):
                             valid_weight = np.multiply(valid_weight, valid_class_sample_weight)
                     valid_init_score = _get_meta_data(eval_init_score, 'eval_init_score', i)
                     valid_group = _get_meta_data(eval_group, 'eval_group', i)
-                    valid_set = _construct_dataset(valid_data[0], valid_data[1],
-                                                   valid_weight, valid_init_score, valid_group, params)
+                    valid_set = Dataset(data=valid_data[0], label=valid_data[1], weight=valid_weight,
+                                        group=valid_group, init_score=valid_init_score,
+                                        categorical_feature='auto', params=params)
```

Comment:
Shouldn't we do something like `reference=train_set` or `train_set.create_valid` here?

Reply:
yeah, I think you're right!

It looks like this logic (the choice to not set this up as a validation set based on the training data) has been there since #113.

I think it's ok to bring that change into the scope of this PR.

Reply:
Ah actually... @jmoralez I think it might not be necessary to do here? Because `.set_reference(train_set)` eventually gets called on every validation set, over in `train()`.

https://github.com/microsoft/LightGBM/blob/1e92ec9e48d76dfd39334a32898ba088648cf790/python-package/lightgbm/engine.py#L188

Reply:
Nice. Thanks for the explanation!